tests/apps/utils/tls.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This utility adapted from https://go.dev/src/crypto/tls/generate_cert.go
+
+package utils
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"log"
+	"math/big"
+	"net"
+	"os"
+	"path"
+	"strings"
+	"time"
+)
+
+const (
+	// clockSkew is the margin of error for checking whether a certificate is valid.
+	clockSkew = time.Minute * 5
+)
+
+// GenerateTLSCertAndKey generates a self-signed X.509 certificate for a TLS server.
+// Outputs to 'cert.pem' and 'key.pem' and will overwrite existing files.
+//
+// host: Comma-separated hostnames and IPs to generate a certificate for
+// validFrom: The time the certificate is valid from
+// validFor: The duration the certificate is valid for
+// directory: Path to write the files to
+func GenerateTLSCertAndKey(host string, validFrom time.Time, validFor time.Duration, directory string) error {
+	// *********************
+	// Generate private key
+	// *********************
+	tlsKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return err
+	}
+
+	b, err := x509.MarshalPKCS8PrivateKey(tlsKey)
+	if err != nil {
+		log.Printf("Unable to marshal ECDSA private key: %v", err)
+		return err
+	}
+
+	if err = bytesToPemFile(path.Join(directory, "key.pem"), "EC PRIVATE KEY", b); err != nil {
```

Comment:
Note that if you change it to use PKCS#8, you need to change the PEM block type to `PRIVATE KEY` (no "EC")

Reply:
Maybe we can unify key strings and put them into `util` pkg:
```
pkg/credentials/tls_test.go:32:var TestKey = `-----BEGIN EC PRIVATE KEY-----
pkg/credentials/tls_test.go:36:-----END EC PRIVATE KEY-----`
pkg/runtime/security/security.go:20:    ecPKType = "EC PRIVATE KEY"
pkg/sentry/ca/certificate_authority_test.go:49: issuerKey = `-----BEGIN EC PRIVATE KEY-----
pkg/sentry/ca/certificate_authority_test.go:53:-----END EC PRIVATE KEY-----`
pkg/sentry/certs/certs.go:17:   BlockTypeECPrivateKey    = "EC PRIVATE KEY"
pkg/sentry/csr/csr.go:22:       blockTypeECPrivateKey = "EC PRIVATE KEY" // EC private key
```

Reply:
No please do not make tests/apps depend on Dapr/Dapr because that will cause a lot of dependency nightmares.

But in general we could do that for all occurrences outside of tests.

Reply:
Thanks @ItalyPaleAle, updated. Now we are using ECDSA + PKCS#8, docs: https://pkg.go.dev/crypto/x509#MarshalPKCS8PrivateKey